### PR TITLE
feat(genqa): add rephrase buttons as an option

### PR DESCRIPTION
**Pull request by jelmedini**
[SVCC-4005](https://coveord.atlassian.net/browse/SVCC-4005)

- Add rephrase buttons as an option.
- By default `withRephraseButtons` is set at `false`.
- When `withRephraseButtons` is set to `true`, we show the rephrase buttons.

By default:

Screenshot 2024-08-01 at 3 14 46 PM

When `withRephraseButtons` is set to `true`:

Screenshot 2024-08-01 at 3 15 08 PM

[SVCC-4005]: https://coveord.atlassian.net/browse/SVCC-4005?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/atomic/cypress/e2e/generated-answer.cypress.ts [hunk 1]
```diff
@@ -85,14 +85,15 @@ describe('Generated Answer Test Suites', () => {
         .init();
     }
 
-    describe('when an answerStyle prop is provided', () => {
+    describe('when an answerStyle and withRephraseButtons props are provided', () => {
```
Comment (fbeaudoincoveo) [inline packages/atomic/cypress/e2e/generated-answer.cypress.ts:88]:
```suggestion
    describe('when an answerStyle prop is provided and withRephraseButtons prop is set to true', () => {
```

Comment (fbeaudoincoveo) [reply packages/atomic/cypress/e2e/generated-answer.cypress.ts:88]: But why is it important to do those tests with `with-rephrase-buttons`?

File: packages/atomic/cypress/e2e/generated-answer.cypress.ts [hunk 2]
```diff
@@ -420,6 +417,26 @@ describe('Generated Answer Test Suites', () => {
             GeneratedAnswerAssertions.assertLogCopyGeneratedAnswer();
           });
         });
+      });
+
+      describe('when withRephraseButtons option is provided', () => {
```
Comment (fbeaudoincoveo) [inline packages/atomic/cypress/e2e/generated-answer.cypress.ts:422]:
```suggestion
      describe('when the withRephraseButtons prop is set to true', () => {
```

**Review by fbeaudoincoveo — COMMENTED** [review]
I believe this should be considered a breaking change as the rephrase buttons are currently being shown by default. So normally we would introduce this in a major version.

Was the decision to hide the buttons by default made by UX?

Is there a plan to communicate this change to live customers?

**Review by louis-bompart — DISMISSED** [review]
Our cypress tests are frozen, we do not accept new tests with Cypress in Atomic.
Please do them with Playwright.
DXUI can offer guidance on Slack if required.

**Comment by jelmedini** [comment]
> I believe this should be considered a breaking change as the rephrase buttons are currently being shown by default. So normally we would introduce this in a major version.
> 
> Was the decision to hide the buttons by default made by UX?
> 
> Is there a plan to communicate this change to live customers?

No not by UX, Yes there is a plan to communicate when they upgrade to this version, @MathieuLavoieSabourin I will let you explain better the purpose of this change

File: packages/atomic/cypress/e2e/generated-answer.cypress.ts (hunk 1 above)
Comment (jelmedini) [reply packages/atomic/cypress/e2e/generated-answer.cypress.ts:88]: because these tests are a combination of the 2 options

**Comment by jelmedini** [comment]
> I believe this should be considered a breaking change as the rephrase buttons are currently being shown by default. So normally we would introduce this in a major version.
> 
> Was the decision to hide the buttons by default made by UX?
> 
> Is there a plan to communicate this change to live customers?

@louis-bompart @fbeaudoincoveo this is a first iteration for the rephrase buttons, they will be completely removed in the next weeks, so I just fixed the cypress tests with the current change.
The decison to remove it i will let @MathieuLavoieSabourin explain more, But we agreed to do a first iteration as an option and not completely remove it. then afterwards, after communicating with clients we will not use anymore the rephrase options.

**Comment by MathieuLavoieSabourin** [comment]
> > I believe this should be considered a breaking change as the rephrase buttons are currently being shown by default. So normally we would introduce this in a major version.
> > Was the decision to hide the buttons by default made by UX?
> > Is there a plan to communicate this change to live customers?
> 
> @louis-bompart @fbeaudoincoveo this is a first iteration for the rephrase buttons, they will be completely removed in the next weeks, so I just fixed the cypress tests with the current change. The decison to remove it i will let @MathieuLavoieSabourin explain more, But we agreed to do a first iteration as an option and not completely remove it. then afterwards, after communicating with clients we will not use anymore the rephrase options.

@louis-bompart @fbeaudoincoveo you can read more about the reasoning behind hiding the rephrase button here : [RGA Component updates Q3 2024](https://coveord.atlassian.net/wiki/spaces/PRODUCT/pages/4295524353/RGA+Component+updates+Q3+2024#Make-rich-formatting-the-only-option)

We don't consider this PR alone a breaking change because of the following : 

- Usage of the Rephrase buttons is very low, users mostly use it to try to get an answer when there is none, but we will also get rid of the "sorry..." message so this will stop

- We aren’t completely removing the Rephrase buttons at this point, just hiding them by default

- The Rephrase buttons can still be shown/used via the withRephraseButtons option

Finally, [enablement](https://coveord.atlassian.net/browse/GLOBAL-99) is aware and will coordinate the client facing communications!

**Comment by fbeaudoincoveo** [comment]
> We don't consider this PR alone a breaking change because of the following :
> 
> * Usage of the Rephrase buttons is very low, users mostly use it to try to get an answer when there is none, but we will also get rid of the "sorry..." message so this will stop

This does seem to justify the change from a product development perspective. Customers may not see it that way, though. That's why we normally introduce breaking changes in major versions: major versions are expected to be breaking / non-backward compatible, and the customer can decide when they're ready migrate to the new version.

> * We aren’t completely removing the Rephrase buttons at this point, just hiding them by default

Modifying the default value of the option _is_ breaking, though: if a customer relying on the default option value wants to keep the same behavior as before, they will have to adjust their code accordingly (i.e., explicitly set the option to true).

It may be alright if there's a fairly small number of customers using the feature and all of them can be convinced in a timely fashion (i.e., before the change is deployed to production) that this is for the best, and are all onboard to deal with the breaking change in a minor version.

It's important however that the change be communicated and agreed upon before the change is deployed to production, because otherwise customer implementations relying on the CDN and targeting v2 without a minor version number will automatically get the breaking change, which could, in worst case scenarios, break their live site if they have custom code relying on the fact that the feature is active by default.

> * The Rephrase buttons can still be shown/used via the withRephraseButtons option

This does not make the change any less breaking. If all customers are onboard to stop using the feature, and we don't want any new implementaions to use it, we might as well remove it right now altogether.

> Finally, [enablement](https://coveord.atlassian.net/browse/GLOBAL-99) is aware and will coordinate the client facing communications!

👍

@MathieuLavoieSabourin see my comments above.

In short, I don't personally mind the change, and it does seem to be for the best, but it's quite unusual to introduce such a breaking change in a minor version, and should therefore be handled extremely carefully IMO.

**Comment by dmgauthier** [comment]
The code looks fine, although maybe more care should be put into the tests, even if it means re-writing them in playwright. I wont fight over this.

But I agree that overall, changing the default behavior of a client side component when they did not explicitly changed their setup **is** breaking. Not that its a bad change, as it has been well though of. But I think its important we all agree with this fact, a breaking change is a breaking change.

Is there any documented guidelines for breaking changes in the UI-kit? Or we just assume general knowledge of the concept ?

**Review by dmgauthier — APPROVED** [review]

**Comment by fbeaudoincoveo** [comment]
> Is there any documented guidelines for breaking changes in the UI-kit? Or we just assume general knowledge of the concept ?

@dmgauthier I'm not aware of official guidelines per-se.

In general, we only introduce breaking changes to generally available features in major versions.

In the case of features that are in a "beta" state (or similar) state, where it is implicitly or explicitly acknowledged by early adopters that they're using the feature at their own risk, and breaking changes are likely to be introduced in the current major version, we'd communicate the changes to the customers we know are using the feature, but would not otherwise restrict ourselves from deploying them to production.

For instance, this is what we did with the headless commerce sub-package until it became officially GA in July.

**Comment by jelmedini** [comment]
> The code looks fine, although maybe more care should be put into the tests, even if it means re-writing them in playwright. I wont fight over this.
> 
> But I agree that overall, changing the default behavior of a client side component when they did not explicitly changed their setup **is** breaking. Not that its a bad change, as it has been well though of. But I think its important we all agree with this fact, a breaking change is a breaking change.
> 
> Is there any documented guidelines for breaking changes in the UI-kit? Or we just assume general knowledge of the concept ?

@dmgauthier it' s because we are removing this feature completely after, That' s why I fixed the current tests and  not adding this change in Playwright as it will just be deleted after anyway.

**Review by fbeaudoincoveo — APPROVED** [review]
